src/yarpWholeBodyStates.cpp

```diff
@@ -1154,23 +1202,52 @@ bool yarpWholeBodyEstimator::computeWorldRootRotoTranslation(double *q_temp)
       }
       return(true);
   }
-  else
-    
+  else    
     return(false);
 }
-
-bool yarpWholeBodyEstimator::setWorldBaseLinkName(std::string linkName)
+bool yarpWholeBodyEstimator::computeBaseVelocity(double* qj,double* dqj)
 {
+  
   if(wholeBodyModel!=NULL)
   {
-    yInfo()<<"Reference link set as world was "<<robot_reference_frame_link;
-    wholeBodyModel->getFrameList().idToIndex(linkName.c_str(),robot_reference_frame_link);
-    yInfo()<<", now it is set to "<<robot_reference_frame_link<<"\n\n\n";
+    int dof = estimates.lastQ.size();
+   // new (&dqjVect) Eigen::Map<Eigen::VectorXd>(dqj,dof);
+    
+    Eigen::Map<Eigen::VectorXd> dqjVect(dqj,dof);
+    Eigen::Matrix<double,6,Eigen::Dynamic,Eigen::RowMajor> complete_jacobian(6,dof+6), joint_jacobian(6,dof),floatingBase_jacobian(6,6);
+    Eigen::Matrix<double,6,Eigen::Dynamic,Eigen::RowMajor> tempMatForComputation(6,dof);//, minusTemp(6,dof);
+    Eigen::VectorXd dvbVect(6);
+    Eigen::Map<Eigen::VectorXd> rotationalVelocityWrapper(estimates.lastBaseVel.data(), estimates.lastBaseVel.size());
```

Comment:
Update: last week with @francesco-romano we discovered that a dynamic allocation like that that was left in the COM computation method in iDynTree was doubling by itself the computational load of the `torqueBalancing` module. This was fixed in https://github.com/robotology-playground/idyntree/commit/390aed5c86f67dad31762a7cc34874cfc3a6801d , but given that this estimation will be used by `torqueBalancing` we should try to get rid of this dynamic memory allocations.